docs/postmortems/2024-02-05-deprecated-icon-rendering.md

```diff
@@ -0,0 +1,113 @@
+---
+date: 2024-02-05
+authors: @literat
+---
+
+# Excessive deprecation logging while rendering Icon component
+
+**Summary**
+
+<!-- What is a one or two-line summary of the event that occurred? -->
+
+The `@lmc-eu/spirit-web-react@1.7.0` is shipped with a new deprecation warning about the `Icon` component.
+
+**Impact**
+
+<!-- What was the scope of impact from the event occurring? How many teams were
+impacted? -->
+
+The teams that use the ` @lmc-eu/spirit-web-react@1.7.0`` and above are impacted by this deprecation warning.
+ `Since `many components use` the `Icon` component, the impact should be extensive in logs that appear in logs both in browser developer tools and in the server logs.
+
+**Root causes**:
+
+<!-- Looking back, what ended up being the main reasons why this event occurred?
+-->
+
+These deprecations should solve the problem with server-side rendering in the Next.js framework where the Icons were not rendered properly. The deprecation warning was added to the `Icon` component to inform the developers that the `Icon` component will use an `html-react-parser`. A developer should install this dependency to make the `Icon` component work properly. This dependency will parse included SVG and render them directly to the DOM. After this, the usage of the `dangerouslySetInnerHTML` will be obsolete. For the condition of whether to use the `Icon` rendered by `html-react-parser` or the `dangerouslySetInnerHTML` was created a new utility called `htmlParser` dynamically imports the library and returns the function or nothing. This solution was not working properly. Also, the deprecation warning appears in both cases of the usage of the `Icon` component.
+
+**Detection**
+
+<!-- How did we find out or discover that this event had occurred? -->
+
+This problem was detected multiple times by the developers and the support question was raised in our Slack channel:
+
+- https://almamedia.slack.com/archives/C068XPSDWQN/p1703704125308169
+- https://almamedia.slack.com/archives/C068XPSDWQN/p1707125126127969
+- https://almamedia.slack.com/archives/C068XPSDWQN/p1707215641782159
+
+Also, the message appears in the browser console and the server logs:
+
+```
+Warning: Icon component is not supported in SSR without use of `html-react-parser`. Please install missing peer dependency.
+```
+
+**Resolution**
+
+<!-- How did we end up addressing this event to mitigate the impact? -->
+
+The deprecation warning was removed as unnecessary in the first place to remove that excessive logging.
+Secondly, the `html-react-parser` was added as a regular dependency and part of the `Icon` `component which `uses`the`dangerouslySetInnerHTML` was removed.
```

Comment:
maybe turn this into steps?
1. ...deprecation warning...
2. ...HTML parser...
3. ...dangerously...